modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/PojosModelValidator.java

```diff
@@ -0,0 +1,123 @@
+        validatePojoHasDefaultConstructor(owner, problemsReporter);
+      }
+    }.walk(extensionModel);
+  }
+
+  private void iterateParametersAndValidate(ParameterizedModel model, ProblemsReporter problemsReporter,
+                                            TypeValidator<ParameterizedModel, Type, ProblemsReporter> validator) {
+    model.getAllParameterModels().forEach(parameterModel -> {
+
+      Optional<ExtensionParameterDescriptorModelProperty> modelProperty =
+          parameterModel.getModelProperty(ExtensionParameterDescriptorModelProperty.class);
+      if (modelProperty.isPresent()) {
+        Type type = modelProperty.get().getExtensionParameter().getType();
+        validator.validateType(model, type, problemsReporter);
+      }
+    });
+  }
+
+  private void validateOverridesEqualsAndHashCode(ParameterizedModel model, ProblemsReporter reporter) {
+    iterateParametersAndValidate(model, reporter, (parameterizedModel, type, problemReporter) -> {
+      ClassInformationAnnotation classInformationAnnotation = type.getClassInformation();
+      if (!classInformationAnnotation.isInterface() && !classInformationAnnotation.isAbstract()
+          && type.asMetadataType() instanceof ObjectType) {
+        Optional<MethodElement> equals = type.getMethod("equals", Object.class);
+        Optional<MethodElement> hashCode = type.getMethod("hashCode");
+        if (!equals.isPresent() || !hashCode.isPresent()) {
+          reporter
+              .addError(new Problem(model,
+                                    format("Type '%s' must override equals and hashCode",
+                                           type.getName())));
+        }
+      }
+    });
+  }
+
+  private void validatePojoHasDefaultConstructor(ParameterizedModel owner, ProblemsReporter problemsReporter) {
+    iterateParametersAndValidate(owner, problemsReporter, (parameterizedModel, type, problemReporter) -> {
+      if (type.isSameType(TypedValue.class)) {
+        type = getFirstGenericType(type).orElse(null);
+      }
+      if (type != null && type.asMetadataType() instanceof ObjectType) {
+        ClassInformationAnnotation classInformationAnnotation = type.getClassInformation();
+        if (!classInformationAnnotation.isAbstract() && !classInformationAnnotation.isInterface()
+            && !classInformationAnnotation.hasDefaultConstructor()) {
+          problemsReporter
+              .addError(new Problem(parameterizedModel, format("Type '%s' does not have a default constructor", type.getName())));
+        }
+      }
+    });
+  }
+
+  private Optional<Type> getFirstGenericType(Type typeWithGenerics) {
+    return !CollectionUtils.isEmpty(typeWithGenerics.getGenerics())
+        ? Optional.ofNullable(typeWithGenerics.getGenerics().get(0).getConcreteType()) : Optional.empty();
+  }
+
+  @FunctionalInterface
+  private interface TypeValidator<M, T, PR> {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Rename this generic name to match the regular expression '^[A-Z][0-9]?$'. ([squid:S00119](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS00119))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7615&issues=AWjKagnwotIXnmsUX912&open=AWjKagnwotIXnmsUX912)

[ISSUE_KEY]: AWjKagnwotIXnmsUX912